### PR TITLE
add: --fix & automated re-install

### PR DESCRIPTION
**Pull request by ericpaulsen**
Hey @bpmct - feel free to take a look when you have the time. I added the `--fix` flag to save the `helm values` to a file and include them in the subsequent re-install. 

I've included `sleep 25` after `helm uninstall` to make sure the resources are deleted before running `helm install`. I'll look into making this a loop to continually check on the resources being deleted in the next iteration.

Since we recommend in `docs` to only upgrade one version at a time, I'm thinking of removing the `Enter version` prompt and write a command get the current version and save it to a variable for use in the latter parts of the script.

Thanks for your help so far!

### CODE REVIEW

**Comment by bpmct** [comment]
> Since we recommend in docs to only upgrade one version at a time, I'm thinking of removing the Enter version prompt and write a command get the current version and save it to a variable for use in the latter parts of the script.

I still like the idea of manually adding a version, as this can be used to easily switch to a release candidate :)

**Comment by bpmct** [comment]
This is perfect!! The loop to watch it would be overkill haha. I just added some (pretty descriptive) commits with some fixes. Feel free to review and merge if you don't have any more changes!

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+tmp/
```
Comment (bpmct) [inline .gitignore:1]: this way, git doesn't pick up the folder

File: upgrade.sh [hunk 2]
```diff
@@ -7,51 +7,102 @@
...
+                        helm upgrade --namespace $NAMESPACE --atomic \
+                        --wait --install --force --version $VERSION \
+                        coder coder/coder --values tmp/current-values.yml   
+                    fi
+                }
+
+                if fix; then
+                    echo "\n\nUpgrade successful."
+                else
+                    echo "\n\nUpgrade failed. Please reference our documentation for troubleshooting
```
Comment (bpmct) [inline upgrade.sh:49]: makes the message a bit easier to see with a couple newlines

File: upgrade.sh [hunk 3]
```diff
@@ -7,51 +7,102 @@
...
+
+            function main() {
+                prerequisites
+
+                if upgrade; then
+                    echo "\n\nUpgrade successful."
+                else
+                    echo "\n\nUpgrade failed. Please reference our documentation for troubleshooting
+                    a failed upgrade https://coder.com/docs/setup/updating#fixing-a-failed-upgrade. 
+                    Alternatively, run ./upgrade.sh --fix to attempt the commands automatically."
```
Comment (bpmct) [inline upgrade.sh:102]: this is awesome! i love how you did everything in functions. makes it very easy to execute and check

File: upgrade.sh [hunk 4]
```diff
@@ -7,51 +7,102 @@
...
+if [ $# -ne 0 ]; then
+    while [ $# -ne 0 ]; do
+        case "$1" in
+            --fix)
+                echo "Attempting to fix your failed upgrade..."
+                echo "Exporting your helm chart values into a file > tmp/current-values.yml..."
+                mkdir -p tmp && helm get values --namespace $NAMESPACE coder > tmp/current-values.yml
+                echo "Uninstalling Coder..."
+                helm uninstall --namespace $NAMESPACE coder
+                echo "Waiting for all resources to delete..."
```
Comment (bpmct) [inline upgrade.sh:28]: great way to give the user status

**Review by bpmct — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (ericpaulsen) [reply .gitignore:1]: Love it - Nice add.

File: upgrade.sh (hunk 3 above)
Comment (ericpaulsen) [reply upgrade.sh:102]: Thanks!

**Pull request merged**